pkg/blobstore/local/file_backed_location_record_array.go

```diff
@@ -0,0 +1,88 @@
+package local
+
+import (
+	"encoding/binary"
+	"sync"
+
+	"github.com/buildbarn/bb-storage/pkg/filesystem"
+)
+
+func serializeLocationRecord(locationRecord LocationRecord) []byte {
+	serialised := make([]byte, 60)
+	for i := 0; i < 32; i++ {
+		serialised[i] = locationRecord.Key.Digest[i]
+	}
+	binary.LittleEndian.PutUint32(serialised[32:36], locationRecord.Key.Attempt)
+	binary.LittleEndian.PutUint64(serialised[36:44], uint64(locationRecord.Location.BlockID))
+	binary.LittleEndian.PutUint64(serialised[44:52], uint64(locationRecord.Location.OffsetBytes))
+	binary.LittleEndian.PutUint64(serialised[52:60], uint64(locationRecord.Location.SizeBytes))
+
+	return serialised
+}
+
+func deserializeLocationRecord(record []byte) LocationRecord {
+	var keyDigest [32]byte
+	copy(keyDigest[0:32], record[0:32])
+	keyAttempt := binary.LittleEndian.Uint32(record[32:36])
+	locationBlockID := int(binary.LittleEndian.Uint64(record[36:44]))
+	locationOffset := int64(binary.LittleEndian.Uint64(record[44:52]))
+	locationSize := int64(binary.LittleEndian.Uint64(record[52:60]))
+
+	key := LocationRecordKey{
+		Digest:  keyDigest,
+		Attempt: keyAttempt,
+	}
+
+	location := Location{
+		BlockID:     locationBlockID,
+		OffsetBytes: locationOffset,
+		SizeBytes:   locationSize,
+	}
+
+	return LocationRecord{
+		Key:      key,
+		Location: location,
+	}
+}
+
+type fileBackedLocationRecordArray struct {
+	recordFile filesystem.FileReadWriter
+	lock       sync.Mutex
```

Comment:
I don't think you need any locking on this, right? Don't the levels above already provide a sufficient amount of locking?

Reply:
I was reasonably sure that there was already enough locking to be safe without adding one here, but I erred on the side of safety. I'll remove the locks here, if they're not needed.